src/rocksdb/tests/basic.rs

```diff
@@ -123,6 +128,127 @@ async fn basic() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[mz_ore::test(tokio::test)]
+#[cfg_attr(miri, ignore)] // unsupported operation: can't call foreign function `rocksdb_create_default_env` on OS `linux`
+async fn associative_merge_operator_test() -> Result<(), anyhow::Error> {
+    // If the test aborts, this may not be cleaned up.
+    let t = tempfile::tempdir()?;
+
+    // A simple merge operator that adds all merge values to the existing key
+    // value, if any.
+    fn merge(
+        _key: &[u8],
+        existing_value: Option<u32>,
+        operands: ValueIterator<bincode::DefaultOptions, u32>,
+    ) -> Option<u32> {
+        let mut val = existing_value.unwrap_or(0);
+
+        for op in operands {
+            val += op;
+        }
+        Some(val)
+    }
+
+    let static_options = InstanceOptions::new(
+        rocksdb::Env::new()?,
+        2,
+        Some(("test".to_string(), merge)),
+        bincode::DefaultOptions::new(),
+    );
+
+    let mut instance = RocksDBInstance::<String, u32>::new(
+        t.path(),
+        static_options,
+        RocksDBConfig::new(Default::default(), None),
+        shared_metrics_for_tests()?,
+        instance_metrics_for_tests()?,
+    )
+    .await?;
+
+    let mut rolling_sum = 50;
+    let key = "a".to_string();
+    // Set the key to an initial value using Put
+    instance
+        .multi_update(vec![(key.clone(), KeyUpdate::Put(rolling_sum))])
+        .await?;
+
+    // Send a bunch of merge operations to the key, each merge transaction
+    // should add 25 million to the key value, and every 10 transactions
+    // we use a Put to set the key to a specific value.
+    for i in 1..100 {
+        let merges = vec![5u32, 5_000_000];
+        rolling_sum += merges.iter().sum::<u32>();
+
+        instance
+            .multi_update(
+                merges
+                    .into_iter()
+                    .map(|v| (key.clone(), KeyUpdate::Merge(v))),
+            )
+            .await?;
+
+        if i % 10 == 0 {
+            rolling_sum += 3;
+            instance
+                .multi_update(vec![(key.clone(), KeyUpdate::Put(rolling_sum))])
+                .await?;
+        }
+    }
+
+    let mut ret = vec![Default::default(); 1];
+    instance
+        .multi_get(vec![key.clone()], ret.iter_mut(), |value| value)
+        .await?;
+
+    // Validate the Get operation returns the correct sum
```

Comment:
```suggestion
    // Validate the Get operation returns the correct sum (which is the final put)
```

Reply:
I'll fix this test instead -- the final count was meant to be based on merge operations, not the final Put value.